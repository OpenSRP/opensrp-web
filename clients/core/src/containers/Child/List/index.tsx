import reducerRegistry from '@onaio/redux-reducer-registry';
import { some } from 'lodash';
import * as React from 'react';
import { connect } from 'react-redux';
import { Store } from 'redux';

/** Display the client list  */
class ClientList extends React.Component<{}, {}> {
    public render() {
        return <div> child list </div>;
    }
}

export { ClientList };

/** Map props to state  */
const mapStateToProps = (state: Partial<Store>) => {
    return {};
};

/** map props to actions */
const mapDispatchToProps = {};

/** connect clientsList to the redux store */
const ConnectedChildList = connect(mapStateToProps, mapDispatchToProps)(ClientList);

export default ConnectedChildList;
