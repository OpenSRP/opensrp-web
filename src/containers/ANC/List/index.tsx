import reducerRegistry from '@onaio/redux-reducer-registry';
import { some } from 'lodash';
import * as React from 'react';
import { connect } from 'react-redux';
import { Table, Row, Col } from 'reactstrap';
import { Store } from 'redux';
import Loading from '../../../components/page/Loading';
import { OPENSRP_CLIENT_ENDPOINT, PAGINATION_SIZE } from '../../../configs/env';
import { OpenSRPService } from '../../../services/opensrp';
import clientsReducer, {
  Client,
  fetchClients,
  getClientsArray,
  reducerName as clientsReducerName,
} from '../../../store/ducks/clients';
import './ancStyle.css';
import SearchBox from '../../../components/page/SearchBox/SearchBox';
import LocationBreadcrumb from '../../../components/page/Breadcrumb';

/** props Interface for the clientList component */
export interface ANCListProps {
  opensrpService: typeof OpenSRPService;
}

/** default props for the clientList component */
export const defaultANCListProps: ANCListProps = {
  opensrpService: OpenSRPService,
};

/** Display the client list  */
class ANCList extends React.Component<ANCListProps, any> {
  public static defaultProps: ANCListProps = defaultANCListProps;

  constructor(props: ANCListProps) {
    super(props);
    this.state = {
      totalRecords: 0,
      ancArray: [],
    };
  }

  public async componentDidMount() {
    const params = {
      clientType: 'anc',
      pageNumber: '1',
      pageSize: PAGINATION_SIZE,
    };
    const { opensrpService } = this.props;
    const ancService = new opensrpService('client/searchByCriteria');
    const response = await ancService.list(params);
    console.log('child total --> ', response.total);
    this.setState({
      totalRecords: response.total,
      ancArray: response.clients,
    });
  }

  public render() {
    /** render loader if there are no clients in state */

    if (!some(this.state.ancArray)) {
      return <Loading />;
    }
    return (
      <div>
        <h3 className="household-title"> ANC ({this.state.totalRecords})</h3>
        {/* <Row>
          <Col md={5}>  
            <div className="household-search-bar">
              <SearchBox searchCallBack={() => {}} placeholder={`Search Client`} />
            </div>
          </Col>
        </Row> */}
        <LocationBreadcrumb />
        <Table striped={true}>
          <thead>
            <tr>
              <th>Identifier</th>
              <th>First Name</th>
              <th>Last Name</th>
              <th>Location</th>
              <th>Age</th>
              <th>EDD</th>
              <th>Gestational age</th>
              <th>Last contact</th>
              <th>Risk category</th>
              <th>Actions</th>
            </tr>
          </thead>
          <tbody>
            {this.state.ancArray.map((client: Client) => {
              return (
                <tr key={client.id}>
                  <td>{client.id}</td>
                  <td>{client.firstName}</td>
                  <td>{client.lastName}</td>
                  <td></td>
                  <td>{client.attributes.dynamicProperties.age_year_part}</td>
                  <td></td>
                  <td></td>
                  <td>{client.attributes.dynamicProperties.last_contact_date.split('T')[0]}</td>
                  <td></td>
                  <td>
                    <a href={`${'#'}`}> view </a>
                  </td>
                </tr>
              );
            })}
          </tbody>
        </Table>
      </div>
    );
  }
}

export { ANCList };
/** Maybe define default props */
/** connect the component to the store */

/** Interface to describe props from mapStateToProps */
interface DispatchedStateProps {
  ancArray: Client[];
}

/** Map props to state  */
const mapStateToProps = (state: Partial<Store>): DispatchedStateProps => {
  const result = {
    ancArray: [],
  };
  return result;
};

/** map props to actions */
const mapDispatchToProps = { fetchClientsActionCreator: fetchClients };

/** connect clientsList to the redux store */
const ConnectedANCList = connect(mapStateToProps, mapDispatchToProps)(ANCList);

export default ConnectedANCList;
