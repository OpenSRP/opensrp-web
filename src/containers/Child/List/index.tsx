import reducerRegistry from '@onaio/redux-reducer-registry';
import { some } from 'lodash';
import * as React from 'react';
import { connect } from 'react-redux';
import { Col, Row, Table, FormGroup, Label, Input } from 'reactstrap';
import { Store } from 'redux';
import LocationBreadcrumb from '../../../components/page/Breadcrumb';
import Loading from '../../../components/page/Loading';
import SearchBox from '../../../components/page/SearchBox/SearchBox';
import { OPENSRP_CLIENT_ENDPOINT, PAGINATION_SIZE } from '../../../configs/env';
import { OpenSRPService } from '../../../services/opensrp';
import clientsReducer, {
  Client,
  fetchClients,
  getClientsArray,
  reducerName as clientsReducerName,
} from '../../../store/ducks/clients';
import './childStyle.css';

/** props Interface for the clientList component */
export interface ChildListProps {
  opensrpService: typeof OpenSRPService;
}

/** default props for the clientList component */
export const defaultChildListProps: ChildListProps = {
  opensrpService: OpenSRPService,
};

/** Display the client list  */
class ChildList extends React.Component<ChildListProps, any> {
  public static defaultProps: ChildListProps = defaultChildListProps;

  constructor(props: ChildListProps) {
    super(props);
    this.state = {
      totalRecords: 0,
      childArray: [],
    };
  }

  public async componentDidMount() {
    const params = {
      clientType: 'child',
      pageNumber: '1',
      pageSize: PAGINATION_SIZE,
    };
    const { opensrpService } = this.props;
    const childService = new opensrpService('client/searchByCriteria');
    const response = await childService.list(params);
    console.log('child total --> ', response.total);
    this.setState({
      totalRecords: response.total,
      childArray: response.clients,
    });
  }

  public render() {
    /** render loader if there are no clients in state */

    if (!some(this.state.childArray)) {
      return <Loading />;
    }
    return (
      <div>
        <h3 className="household-title"> Child ({this.state.totalRecords})</h3>
        <Row>
          <Col md={5}>
            <div className="household-search-bar">
              <SearchBox searchCallBack={() => {}} placeholder={`Search Client`} />
            </div>
          </Col>
          <Col md={3} style={{ marginTop: '23px' }}>
            <FormGroup>
              <Input
                type="select"
                name="select"
                className="shadow-sm"
                style={{ fontSize: '12px', borderRadius: '1px' }}
              >
                <option value="">Select gender</option>
                <option value="Male">Male</option>
                <option value="Female"> Female</option>
              </Input>
            </FormGroup>
          </Col>
        </Row>
        <Row>
          <Col>
            <LocationBreadcrumb />
            <Table striped={true}>
              <thead>
                <tr>
                  <th>Identifier</th>
                  <th>First Name</th>
                  <th>Last Name</th>
                  <th>Location</th>
                  <th>Age</th>
                  <th>Gender</th>
                  <th>Last contact date</th>
                  <th>Risk</th>
                  <th>Immunization status</th>
                  <th>Actions</th>
                </tr>
              </thead>
              <tbody>
                {this.state.childArray.map((client: Client) => {
                  return (
                    <tr key={client.id}>
                      <td>{client.id}</td>
                      <td>{client.firstName}</td>
                      <td>{client.lastName}</td>
                      <td></td>
                      <td>{client.attributes.dynamicProperties.age_year_part}</td>
                      <td>{client.gender}</td>
                      <td>{client.attributes.dynamicProperties.last_contact_date.split('T')[0]}</td>
                      <td></td>
                      <td></td>
                      <td>
                        <a href={`${'#'}`}> view </a>
                      </td>
                    </tr>
                  );
                })}
              </tbody>
            </Table>
          </Col>
        </Row>
      </div>
    );
  }
}

export { ChildList };
/** Maybe define default props */
/** connect the component to the store */

/** Interface to describe props from mapStateToProps */
interface DispatchedStateProps {
  childArray: Client[];
}

/** Map props to state  */
const mapStateToProps = (state: Partial<Store>): DispatchedStateProps => {
  const result = {
    childArray: [],
  };
  return result;
};

/** map props to actions */
const mapDispatchToProps = { fetchClientsActionCreator: fetchClients };

/** connect clientsList to the redux store */
const ConnectedChildList = connect(mapStateToProps, mapDispatchToProps)(ChildList);

export default ConnectedChildList;
