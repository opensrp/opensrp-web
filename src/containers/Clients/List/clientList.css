tr:nth-child(odd) {
  background-color: #f2f2f2;
}

tr:first-child {
  background-color: white;
}

.table {
  border: 1px outset #f2f2f2;
}

.table-shadow {
  box-shadow: 1px 2px #c9c9c9;
}

th .form-control {
  border: none;
  font: inherit;
}
