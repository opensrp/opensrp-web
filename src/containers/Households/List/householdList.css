.household-search-bar{
  margin: 24px 0px 24px 0px;
}

.table {
  background: white;
  border: None;
}

thead {
  color: #3A484F;
  font-size: 12px;
  font-weight: 400;
  line-height: 17px;
  text-align: left;
}

tbody {
  font-size: 12px;
  color: #7c7c7c;
}

.members-table-field{
  color: #007bff;
}

.household-title{
  color: #3A484F;
}

.back-btn-bg{
  font-size: 14px;
}