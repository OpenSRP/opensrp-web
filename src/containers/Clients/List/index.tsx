import reducerRegistry from '@onaio/redux-reducer-registry';
import { some } from 'lodash';
import * as React from 'react';
import { connect } from 'react-redux';
import { Table, Row, Col } from 'reactstrap';
import { Store } from 'redux';
import Loading from '../../../components/page/Loading';
import { OPENSRP_CLIENT_ENDPOINT, PAGINATION_SIZE } from '../../../configs/env';
import { OpenSRPService } from '../../../services/opensrp';
import clientsReducer, {
  Client,
  fetchClients,
  getClientsArray,
  reducerName as clientsReducerName,
} from '../../../store/ducks/clients';
import './clientList.css';
import SearchBox from '../../../components/page/SearchBox/SearchBox';

/** register the clients reducer */
reducerRegistry.register(clientsReducerName, clientsReducer);

/** props Interface for the clientList component */
export interface ClientListProps {
  opensrpService: typeof OpenSRPService;
  clientsArray: Client[];
  fetchClientsActionCreator: typeof fetchClients;
}

/** default props for the clientList component */
export const defaultClientListProps: ClientListProps = {
  clientsArray: [],
  fetchClientsActionCreator: fetchClients,
  opensrpService: OpenSRPService,
};

/** Display the client list  */
class ClientList extends React.Component<ClientListProps, any> {
  public static defaultProps: ClientListProps = defaultClientListProps;

  constructor(props: ClientListProps) {
    super(props);
    this.state = {
      totalRecords: 0,
    }
  }

  public async componentDidMount() {
    const params = {
      clientType: 'clients',
      pageNumber: '1',
      pageSize: PAGINATION_SIZE,
    };
    const { fetchClientsActionCreator, opensrpService } = this.props;
    const clientService = new opensrpService('client/searchByCriteria');
    const response = await clientService.list(params);
    console.log('clients total --> ', response.total);
    fetchClientsActionCreator(response.clients);
    this.setState({ totalRecords: response.total });
  }

  public render() {
    /** render loader if there are no clients in state */
    const { clientsArray } = this.props;
    if (!some(clientsArray)) {
      return <Loading />;
    }
    return (
      <div>
        <h3 className="household-title"> Clients ({this.state.totalRecords})</h3>
        {/* <Row>
          <Col md={5}>
            <div className="household-search-bar">
              <SearchBox searchCallBack={() => {}} placeholder={`Search Client`} />
            </div>
          </Col>
        </Row> */}

        <Table striped={true}>
          <thead>
            <tr>
              <th>Identifier</th>
              <th>First Name</th>
              <th>Middle Name</th>
              <th>Last Name</th>
              <th>Gender</th>
              <th>actions</th>
            </tr>
          </thead>
          <tbody>
            {clientsArray.map((client: Client) => {
              return (
                <tr key={client.id}>
                  <td>{client.id}</td>
                  <td>{client.firstName}</td>
                  <td>{client.middleName}</td>
                  <td>{client.lastName}</td>
                  <td>{client.gender}</td>
                  <td>
                    <a href={`${'#'}`}> view </a>
                  </td>
                </tr>
              );
            })}
          </tbody>
        </Table>
      </div>
    );
  }
}

export { ClientList };
/** Maybe define default props */
/** connect the component to the store */

/** Interface to describe props from mapStateToProps */
interface DispatchedStateProps {
  clientsArray: Client[];
}

/** Map props to state  */
const mapStateToProps = (state: Partial<Store>): DispatchedStateProps => {
  const result = {
    clientsArray: getClientsArray(state),
  };
  return result;
};

/** map props to actions */
const mapDispatchToProps = { fetchClientsActionCreator: fetchClients };

/** connect clientsList to the redux store */
const ConnectedClientList = connect(mapStateToProps, mapDispatchToProps)(ClientList);

export default ConnectedClientList;
