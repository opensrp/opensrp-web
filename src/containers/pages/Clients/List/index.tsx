import reducerRegistry from '@onaio/redux-reducer-registry';
import { some } from 'lodash';
import * as React from 'react';
import { connect } from 'react-redux';
import { Table } from 'reactstrap';
import { Store } from 'redux';
import Loading from '../../../../components/page/Loading';
import { OPENSRP_CLIENT_ENDPOINT } from '../../../../configs/env';
import { OpenSRPService } from '../../../../services/opensrp';
import clientsReducer, {
  Client,
  fetchClients,
  getClientsArray,
  reducerName as clientsReducerName,
} from '../../../../store/ducks/clients';
import './clientList.css';

/** register the clients reducer */
reducerRegistry.register(clientsReducerName, clientsReducer);

/** props Interface for the clientList component */
export interface ClientListProps {
  opensrpService: typeof OpenSRPService;
  clientsArray: Client[];
  fetchClientsActionCreator: typeof fetchClients;
}

/** default props for the clientList component */
export const defaultClientListProps: ClientListProps = {
  clientsArray: [],
  fetchClientsActionCreator: fetchClients,
  opensrpService: OpenSRPService,
};

/** Display the client list  */
class ClientList extends React.Component<ClientListProps, {}> {
  public static defaultProps: ClientListProps = defaultClientListProps;
  constructor(props: ClientListProps) {
    super(props);
  }

  public async componentDidMount() {
    const { fetchClientsActionCreator, opensrpService } = this.props;
    const clientService = new opensrpService(`${OPENSRP_CLIENT_ENDPOINT}`);
    const response = await clientService.list();
    fetchClientsActionCreator(response);
  }

  public render() {
    /** render loader if there are no clients in state */
    const { clientsArray } = this.props;
    if (!some(clientsArray)) {
      return <Loading />;
    }
    return (
      <div>
        <Table striped={true}>
          <thead>
            <tr>
              <th>Identifier</th>
              <th>First Name</th>
              <th>Middle Name</th>
              <th>Last Name</th>
              <th>Gender</th>
              <th>actions</th>
            </tr>
          </thead>
          <tbody>
            {clientsArray.map((client: Client) => {
              return (
                <tr key={client._id}>
                  <td>{client._id}</td>
                  <td>{client.firstName}</td>
                  <td>{client.middleName}</td>
                  <td>{client.lastName}</td>
                  <td>{client.gender}</td>
                  <td>
                    <a href={`${'#'}`}> view </a>
                  </td>
                </tr>
              );
            })}
          </tbody>
        </Table>
      </div>
    );
  }
}

export { ClientList };
/** Maybe define default props */
/** connect the component to the store */

/** Interface to describe props from mapStateToProps */
interface DispatchedStateProps {
  clientsArray: Client[];
}

/** Map props to state  */
const mapStateToProps = (state: Partial<Store>): DispatchedStateProps => {
  const result = {
    clientsArray: getClientsArray(state),
  };
  return result;
};

/** map props to actions */
const mapDispatchToProps = { fetchClientsActionCreator: fetchClients };

/** connect clientsList to the redux store */
const ConnectedClientList = connect(
  mapStateToProps,
  mapDispatchToProps
)(ClientList);

export default ConnectedClientList;
