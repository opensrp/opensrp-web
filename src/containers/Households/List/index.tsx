import reducerRegistry from '@onaio/redux-reducer-registry';
import { some } from 'lodash';
import * as React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import { Col, Row, Table } from 'reactstrap';
import { Store } from 'redux';
import Loading from '../../../components/page/Loading';
import SearchBox from '../../../components/page/SearchBox/SearchBox';
import { OPENSRP_HOUSEHOLD_ENDPOINT, PAGINATION_SIZE } from '../../../configs/env';
import {
  householdClientType as clientType,
  householdSearchPlaceholder as searchPlaceholder,
} from '../../../constants';
import { FlexObject } from '../../../helpers/utils';
import { OpenSRPService } from '../../../services/opensrp';
import householdsReducer, {
  fetchHouseholds,
  getHouseholdsArray,
  getTotalRecords,
  Household,
  reducerName as householdsReducerName,
  removeHouseholds,
  setTotalRecords,
} from '../../../store/ducks/households';
import './householdList.css';

/** register the households reducer */
reducerRegistry.register(householdsReducerName, householdsReducer);

/** props Interface for the householdList component */
export interface HouseholdListProps {
  opensrpService: typeof OpenSRPService;
  householdsArray: Household[];
  totalRecordsCount: number;
  fetchHouseholdsActionCreator: typeof fetchHouseholds;
  setTotalRecordsActionCreator: typeof setTotalRecords;
  removeHouseholdsActionCreator: typeof removeHouseholds;
}

export interface HouseholdListState {
  loading: boolean;
  pageNumber: string;
}

export const defaultHouseholdListState: HouseholdListState = {
  loading: true,
  pageNumber: '1',
};

/** default props for the householdList component */
export const defaultHouseholdListProps: HouseholdListProps = {
  fetchHouseholdsActionCreator: fetchHouseholds,
  householdsArray: [],
  opensrpService: OpenSRPService,
  removeHouseholdsActionCreator: removeHouseholds,
  setTotalRecordsActionCreator: setTotalRecords,
  totalRecordsCount: 0,
};

/** Display the Household list  */
class HouseholdList extends React.Component<HouseholdListProps, HouseholdListState> {
  public static defaultProps: HouseholdListProps = defaultHouseholdListProps;

  public constructor(props: HouseholdListProps) {
    super(props);
    this.state = defaultHouseholdListState;
  }
  public async componentDidMount() {
    this.getHouseholdList();
  }

  public render() {
    /** render loader if there are no households in state */
    const { householdsArray, totalRecordsCount } = this.props;
    if (this.state.loading) {
      return <Loading />;
    } else {
      return (
        <div>
          <h3 className="household-title"> Household ({totalRecordsCount}) </h3>
          <Row>
            <Col md={5}>
              <div className="household-search-bar">
                <SearchBox searchCallBack={this.search} placeholder={searchPlaceholder} />
              </div>
            </Col>
          </Row>
          <Row>
            <Col>
              <Table className="shadow-sm">
                <thead>
                  <tr>
                    <th>HH ID Number</th>
                    <th>Family Name</th>
                    <th>Head of Household</th>
                    <th>Phone</th>
                    <th>Registered Date</th>
                    <th>Members</th>
                    <th>Actions</th>
                  </tr>
                </thead>
                <tbody>
                  {householdsArray.map((household: Household) => {
                    return (
                      <tr key={household.baseEntityId}>
                        <td>{household.baseEntityId}</td>
                        <td>{household.lastName}</td>
                        <td>{household.attributes.HouseholdHead}</td>
                        <td>{household.attributes.phoneNumber}</td>
                        <td>{household.dateCreated}</td>
                        <td className="members-table-field">{household.attributes.memberCount}</td>
                        <td>
                          <Link to={`/household/profile/${household.baseEntityId}/`}>View</Link>
                        </td>
                      </tr>
                    );
                  })}
                </tbody>
              </Table>
            </Col>
          </Row>
        </div>
      );
    }
  }

  private search = (searchText: string) => {
    this.getHouseholdList({ searchText });
  };

  private async getHouseholdList(extraParams: FlexObject = {}) {
    const params = {
      clientType,
      pageNumber: this.state.pageNumber,
      pageSize: PAGINATION_SIZE,
      ...extraParams,
    };
    const {
      fetchHouseholdsActionCreator,
      setTotalRecordsActionCreator,
      removeHouseholdsActionCreator,
      opensrpService,
    } = this.props;
    const hosueholdService = new opensrpService(`${OPENSRP_HOUSEHOLD_ENDPOINT}`);
    const response = await hosueholdService.list(params);
    removeHouseholdsActionCreator();
    this.setState(
      {
        ...this.state,
        loading: false,
      },
      () => {
        fetchHouseholdsActionCreator(response.clients);
        setTotalRecordsActionCreator(response.total);
      }
    );
  }
}

export { HouseholdList };

/** Interface to describe props from mapStateToProps */
interface DispatchedStateProps {
  householdsArray: Household[];
  totalRecordsCount: number;
}

/** Map props to state  */
const mapStateToProps = (state: Partial<Store>): DispatchedStateProps => {
  const result = {
    householdsArray: getHouseholdsArray(state),
    totalRecordsCount: getTotalRecords(state),
  };
  return result;
};

/** map props to actions */
const mapDispatchToProps = {
  fetchHouseholdsActionCreator: fetchHouseholds,
  removeHouseholdsActionCreator: removeHouseholds,
  setTotalRecordsActionCreator: setTotalRecords,
};

/** connect householdsList to the redux store */
const ConnectedHouseholdList = connect(
  mapStateToProps,
  mapDispatchToProps
)(HouseholdList);

export default ConnectedHouseholdList;
