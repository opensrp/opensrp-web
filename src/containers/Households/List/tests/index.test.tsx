import reducerRegistry from '@onaio/redux-reducer-registry';
import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import * as React from 'react';
import { Provider } from 'react-redux';
import store from '../../../../store';
import householdsReducer, {
  reducerName as householdsReducerName,
} from '../../../../store/ducks/households';
import { HouseholdList, HouseholdListProps } from '../../List/index';

reducerRegistry.register(householdsReducerName, householdsReducer);

jest.mock('../../../../configs/env');
describe('containers/households/list/Householdlist', () => {
  beforeEach(() => {
    jest.resetAllMocks();
  });

  it('renders without crashing', () => {
    const props: HouseholdListProps = {
      fetchHouseholdsActionCreator: jest.fn(),
      householdsArray: [],
      opensrpService: jest.fn(),
      removeHouseholdsActionCreator: jest.fn(),
      setTotalRecordsActionCreator: jest.fn(),
      totalRecordsCount: 0,
    };
    shallow(<HouseholdList {...props} />);
  });

  it('works correctly with the redux store', () => {
    const props: HouseholdListProps = {
      fetchHouseholdsActionCreator: jest.fn(),
      householdsArray: [],
      opensrpService: jest.fn(),
      removeHouseholdsActionCreator: jest.fn(),
      setTotalRecordsActionCreator: jest.fn(),
      totalRecordsCount: 0,
    };
    const wrapper = mount(
      <Provider store={store}>
        <HouseholdList {...props} />
        );
      </Provider>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
    wrapper.unmount();
  });
});
