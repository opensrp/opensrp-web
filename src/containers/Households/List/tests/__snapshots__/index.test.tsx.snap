// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/households/list/Householdlist works correctly with the redux store 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <HouseholdList
    fetchHouseholdsActionCreator={[MockFunction]}
    householdsArray={Array []}
    opensrpService={
      [MockFunction] {
        "calls": Array [
          Array [
            "undefined",
          ],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      }
    }
    removeHouseholdsActionCreator={[MockFunction]}
    setTotalRecordsActionCreator={[MockFunction]}
    totalRecordsCount={0}
  >
    <Ripple
      borderColor="#ff5d00"
      borderStyle="solid"
      borderWidth="4px"
      height="64px"
      minHeight="80vh"
      width="64px"
    >
      <div
        className="lds-ripple-parent"
        style={
          Object {
            "minHeight": "80vh",
          }
        }
      >
        <div
          className="lds-ripple"
          style={
            Object {
              "height": "64px",
              "width": "64px",
            }
          }
        >
          <div
            style={
              Object {
                "borderColor": "#ff5d00",
                "borderStyle": "solid",
                "borderWidth": "4px",
              }
            }
          />
          <div
            style={
              Object {
                "borderColor": "#ff5d00",
                "borderStyle": "solid",
                "borderWidth": "4px",
              }
            }
          />
        </div>
      </div>
    </Ripple>
  </HouseholdList>
  );
</Provider>
`;
